pkg/subctl/cmd/gather/gather.go

```diff
@@ -116,54 +116,43 @@ func gatherData(cluster *cmd.Cluster) bool {
 		}
 	}
 
-	gatherDataByCluster(cluster, directory)
-
-	fmt.Printf("Files are stored under directory %q\n", directory)
+	status := cli.NewReporter()
+	status.Start("Gathering information from cluster %q\n", cluster.Name)
+	gatherDataByCluster(cluster, directory, status)
 
+	status.Success("Files are stored under directory %q\n", directory)
+    status.End()
 	return true
 }
 
-func gatherDataByCluster(cluster *cmd.Cluster, directory string) {
+func gatherDataByCluster(cluster *cmd.Cluster, directory string, status reporter.Interface) {
 	var err error
 	clusterName := cluster.Name
 
-	fmt.Printf("Gathering information from cluster %q\n", clusterName)
+	clientProducer, err := client.NewProducerFromRestConfig(cluster.Config)
+	if err != nil {
+		status.Failure("Error creating client producer")
+		status.End()
+		return
```

Comment:
Using `sttaus.Error` means I will have to change `gatherDataByCluster` to return error and check the error at the caller.

```
gatherDataByCluster(cluster, directory, status)

    status.Success("Files are stored under directory %q\n", directory)
    status.End()
```

Checking for error here is not logical because there is no further operation happening based on the error. More logical approach is to log the error via `status.Failure` and move to `status.Success("Files are stored under directory %q\n", directory)`